VitalHealthConnect/src/main/java/io/tryvital/vitalhealthconnect/VitalHealthConnectManager.kt

```diff
@@ -435,70 +444,73 @@ class VitalHealthConnectManager private constructor(
         sharedPreferences.edit().remove(UnSecurePrefKeys.changeTokenKey).apply()
     }
 
-    private fun startWorkerForChanges(userId: String, healthResource: Set<VitalResource>) {
-        val work = WorkManager.getInstance(context).beginUniqueWork(
-            "UploadChangesWorker",
-            ExistingWorkPolicy.REPLACE,
-            OneTimeWorkRequestBuilder<UploadChangesWorker>().setInputData(
-                UploadChangesWorker.createInputData(
-                    userId = userId,
-                    region = vitalClient.region,
-                    environment = vitalClient.environment,
-                    apiKey = vitalClient.apiKey,
-                    resource = healthResource
-                )
-            ).build()
-        )
+    private suspend fun startWorkerForChanges(userId: String, healthResource: Set<VitalResource>) {
+        val workRequest = OneTimeWorkRequestBuilder<UploadChangesWorker>().setInputData(
+            UploadChangesWorker.createInputData(
+                userId = userId,
+                region = vitalClient.region,
+                environment = vitalClient.environment,
+                apiKey = vitalClient.apiKey,
+                resource = healthResource
+            )
+        ).build()
 
-        taskScope.launch(Dispatchers.Main) {
-            val observer = androidx.lifecycle.Observer<List<WorkInfo>> { workInfos ->
-                updateStatusFromJob(workInfos)
-            }
-            work.workInfosLiveData.observeForever(observer)
-            work.enqueue()
+        return executeWork(workRequest, "UploadChangesWorker")
+    }
 
-            try {
-                awaitCancellation()
-            } catch (e: Throwable) {
-                work.workInfosLiveData.removeObserver(observer)
-            }
-        }
+    private suspend fun startWorkerForAllData(userId: String, healthResource: Set<VitalResource>) {
+        val workRequest = OneTimeWorkRequestBuilder<UploadAllDataWorker>().setInputData(
+            UploadAllDataWorker.createInputData(
+                startTime = Instant.now().minus(
+                    encryptedSharedPreferences.getInt(
+                        UnSecurePrefKeys.numberOfDaysToBackFillKey, 30
```

Comment:
Is the default here 30?

Reply:
yeah, but I think it would never be used. This preference key is always written to in configure(_:).